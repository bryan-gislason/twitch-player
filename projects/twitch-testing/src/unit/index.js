import { ThemeProvider } from "@twitch-player/themes/dist/context";
import { light as lightTheme } from "@twitch-player/themes/dist/themes";
import { OverridesProvider } from "@twitch-player/ui/dist/context";
import React from "react";
import { act } from "react-dom/test-utils";
import { MemoryRouter as Router } from "react-router";
import { fireEvent, render } from "react-testing-library";

const customRender = (node, ...options) => {
  return render(
    <ThemeProvider theme={lightTheme}>
      <OverridesProvider>
        <Router>{node}</Router>
      </OverridesProvider>
    </ThemeProvider>,
    ...options
  );
};

fireEvent["press"] = (node, init) => {
  fireEvent.mouseDown(node, init);
  fireEvent.mouseUp(node, init);
};

fireEvent["resize"] = (node, init) => {
  const { target: { left = 0, top = 0, width = 0, height = 0 } = {} } = init;

  Object.defineProperties(window.HTMLElement.prototype, {
    offsetLeft: { get: () => left },
    offsetTop: { get: () => top },
    offsetWidth: { get: () => width },
    offsetHeight: { get: () => height },
  });

  const eventInit = { bubbles: false, cancelable: false };
  const event = new Event(node, eventInit);

  act(() => {
    fireEvent(node, event);
    // eslint-disable-next-line no-undef
    jest.runAllTimers();
  });
};

export * from "react-testing-library";

export { customRender as render };
