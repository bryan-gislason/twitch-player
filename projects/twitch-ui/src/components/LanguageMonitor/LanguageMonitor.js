import { memo } from "react";
import { I18nManager } from "react-native";
import { useLanguage } from "../../hooks";

const LanguageMonitor = memo(() => {
  useLanguage((lng, dir) => {
    document.documentElement.setAttribute("lang", lng);
    I18nManager.forceRTL(dir === "rtl");
  });

  return null;
});

export default LanguageMonitor;
