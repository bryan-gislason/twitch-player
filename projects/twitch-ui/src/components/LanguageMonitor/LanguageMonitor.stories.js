import { storiesOf } from "@storybook/react";
import React from "react";
import LanguageMonitor from ".";

storiesOf("Components|LanguageMonitor", module).add("default", () => (
  <LanguageMonitor />
));
