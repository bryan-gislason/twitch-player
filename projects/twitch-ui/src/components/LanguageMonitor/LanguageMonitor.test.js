import { render } from "@twitch-player/testing/dist/unit";
import React from "react";
import LanguageMonitor from "./LanguageMonitor";

describe("LanguageMonitor", () => {
  let instance;

  beforeEach(() => {
    instance = render(<LanguageMonitor />);
  });

  it("renders nothing", () => {
    expect(instance.asFragment()).toMatchSnapshot();
  });

  it("sets the `lang` attribute of the `<html>` tag", () => {
    expect(document.documentElement.getAttribute("lang")).toBe("en-US");
  });

  it("sets the `dir` attribute of the `<html>` tag", () => {
    expect(document.documentElement.getAttribute("dir")).toBe("ltr");
  });
});
