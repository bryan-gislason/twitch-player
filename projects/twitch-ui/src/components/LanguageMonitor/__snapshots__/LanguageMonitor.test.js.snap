// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageMonitor renders nothing 1`] = `<DocumentFragment />`;
